### DISCUSSION TITLE
concatenated strings as primary keys [Apr 07]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#1682] [Discord Threads]
> 🙋‍♂️ **Marian**

**Comment by o1lab** [comment]
> 👨‍🚀 **wingkwong**  
#### what do you mean by concatenated strings? like two columns with string type? like composite PK ?  
#  
> 🙋‍♂️ **Marian**  
#### Like using a formula column and using concat with 4 columns and make it a PK so i can reference search it  
#  
> 👨‍🚀 **wingkwong**  
#### oh. in this case you cannot, as formula column is a virtual column.  
#  
> 🙋‍♂️ **Marian**  
#### So my best option is to make it via the API and use that column as PK?  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/960357597846839352)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
